docs/framework/data/adonet/retrieving-data-using-a-datareader.md

```diff
@@ -1,131 +1,140 @@
 ---
 title: "Retrieving Data Using a DataReader"
-ms.date: "03/30/2017"
+ms.date: "10/25/2018"
 dev_langs: 
   - "csharp"
   - "vb"
 ms.assetid: 97afc121-fb8b-465b-bab3-6d844420badb
 ---
-# Retrieving Data Using a DataReader
-Retrieving data using a **DataReader** involves creating an instance of the **Command** object and then creating a **DataReader** by calling **Command.ExecuteReader** to retrieve rows from a data source. The following example illustrates using a **DataReader** where `reader` represents a valid DataReader and `command` represents a valid Command object.  
-  
-```  
+# Retrieve Data Using a DataReader
+To retrieve data using a **DataReader**, create an instance of the **Command** object, and then create a **DataReader** by calling **Command.ExecuteReader** to retrieve rows from a data source. The **DataReader** provides an unbuffered stream of data that allows procedural logic to efficiently process results from a data source sequentially. The **DataReader** is a good choice when you're retrieving large amounts of data because the data is not cached in memory.
+
+The following example illustrates using a **DataReader** where `reader` represents a valid DataReader and `command` represents a valid Command object.  
+
+```csharp
 reader = command.ExecuteReader();  
+```
+
+```vb
+reader = command.ExecuteReader()
 ```  
+
+Use the **Read** method of the **DataReader** object to obtain a row from the results of the query. You can access each column of the returned row by passing the name or ordinal reference of the column to the **DataReader**. However, for best performance, the **DataReader** provides a series of methods that allow you to access column values in their native data types (**GetDateTime**, **GetDouble**, **GetGuid**, **GetInt32**, and so on). For a list of typed accessor methods for data provider-specific **DataReaders**, see <xref:System.Data.OleDb.OleDbDataReader> and <xref:System.Data.SqlClient.SqlDataReader>. Using the typed accessor methods when you know the underlying data type reduces the amount of type conversion required when retrieving the column value.  
```

Comment:
Read method of the DataReader object --> DataReader.Read method
from the results of the query --> from the query results.
ordinal reference --> ordinal number